centml/compiler/server.py

```diff
@@ -43,8 +43,12 @@ def background_compile(model_id: str, tfx_graph, example_inputs):
         dir_cleanup(model_id)
 
     try:
+        # torch.save creates an empty zip file then saves the data in multiple calls.
+        # We don't want this incomplete zipfile to be mistaken for the serialized forward function.
         save_path = get_server_compiled_forward_path(model_id)
-        torch.save(compiled_graph_module, save_path, pickle_protocol=config_instance.PICKLE_PROTOCOL)
+        tmp_path = save_path + ".tmp"
+        torch.save(compiled_graph_module, tmp_path, pickle_protocol=config_instance.PICKLE_PROTOCOL)
+        os.rename(tmp_path, save_path)
```

Comment:
can you write a comment why we need to store to a .tmp file first and then rename it? for future reference.